pkg/k8sutil/create_test.go

```diff
@@ -287,3 +288,177 @@ func TestCreateOrUpdateNamespaceUpdateSuccess(t *testing.T) {
 		t.Fatalf("expected %q, got: %q", name, mockns.ObjectMeta.Labels[internal.NamespaceLabelKey])
 	}
 }
+
+func TestYAMLToUnstructured(t *testing.T) {
+	tests := []struct {
+		name    string
+		yamlObj []byte
+		want    *unstructured.Unstructured
+		wantErr bool
+	}{
+		{
+			name: "Valid_Kubernetes_object",
+			yamlObj: []byte(`apiVersion: v1
+kind: Namespace
+metadata:
+  name: test`),
+			want: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "Namespace",
+					"metadata":   map[string]interface{}{"name": "test"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Invalid_Kubernetes_object",
+			yamlObj: []byte(`foobar`),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := YAMLToUnstructured(tt.yamlObj)
+			if (err != nil) != tt.wantErr {
```

Comment:
I think `err != nil && !tt.wantErr` is easier to understand, read as: "error is not nil and I don't want error". Perhaps we could rename `wantErr` to `expectErr`, as it makes more sense then.

EDIT: Ah, this handles both cases when error is expected and not, right? I still wonder if this is a better approach. With 2 separate ifs, we can provide nicer error message.

Reply:
How is error message gonna change? We are just showing boolean values.

Reply:
With 2 separate cases you have simpler logic to grasp + you can have `unexpected error: %v` and `expected error` error messages. But I think simpler logic is more important than error messages in this case.

Reply:
changing it to `if err != nil && !tt.wantErr || err == nil && tt.wantErr {`.